packages/builders/src/components/Assertions.ts

```diff
@@ -1,43 +1,42 @@
 import { APIMessageComponentEmoji, ButtonStyle } from 'discord-api-types/v9';
-import { z } from 'zod';
+import { s } from '@sapphire/shapeshift';
 import type { SelectMenuOptionBuilder } from './selectMenu/SelectMenuOption';
 
-export const customIdValidator = z.string().min(1).max(100);
+export const customIdValidator = s.string.lengthGe(1).lengthLe(100);
 
-export const emojiValidator = z
-	.object({
-		id: z.string(),
-		name: z.string(),
-		animated: z.boolean(),
-	})
-	.partial()
-	.strict();
+export const emojiValidator = s.object({
+	id: s.string,
+	name: s.string,
+	animated: s.boolean,
+}).partial.strict;
 
-export const disabledValidator = z.boolean();
+export const disabledValidator = s.boolean;
 
-export const buttonLabelValidator = z.string().nonempty().max(80);
+export const buttonLabelValidator = s.string.lengthGe(1).lengthLe(80);
 
-export const buttonStyleValidator = z.number().int().min(ButtonStyle.Primary).max(ButtonStyle.Link);
+export const buttonStyleValidator = s.nativeEnum(ButtonStyle);
 
-export const placeholderValidator = z.string().max(150);
-export const minMaxValidator = z.number().int().min(0).max(25);
+export const placeholderValidator = s.string.lengthLe(150);
+export const minMaxValidator = s.number.int.ge(0).le(25);
 
-export const optionsValidator = z.object({}).array().nonempty();
+export const optionsValidator = s.object({}).array.lengthGe(1);
```

Comment:
Why is this an empty object?

By the way, `zod`'s `z.object()` defaults to a passthrough object validator, however, `@sapphire/shapeshift` has stricter defaults, and as such, it defaults to an ignore object validator.

This means, while `[{ foo: 'bar' }]` is valid for this validator, the properties would be stripped and will always result on `[{}]`. If unknown properties are desired to be kept (albeit no validation done for them), you need to use `.passthrough`.

Reply:
s.object is default to ignore so if u pass any key value it'll ignore them and will return `{}`. In this case 
https://github.com/discordjs/discord.js/blob/87a6b8445bfbf3981cd39813fe961dfa1c7f2bce/packages/builders/src/components/Assertions.ts#L29

we don't need the return value. It's just there to check the input is an array of objects. So imo adding `passthrough` here is redundant.